cpp/include/cudf/round_robin.hpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2018-2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <vector>
+#include <cudf/cudf.h>
+#include <cudf/types.hpp>
+
+namespace cudf {
+namespace experimental {
+
+/**
+ * @brief Round-robin partition.
+ * 
+ * Returns a new table with rows re-arranged into partition groups and
+ * a vector of row offsets to the start of each partition in the output table.
+ * Rows are assigned partitions based on their row index in the table,
+ * based on the following formula:
+ * partition = (row_index + start_partition) % num_partitions
+ *
+ * Example 1:
+ * input:
+ * table => col 1 {0, ..., 12}
+ * num_partitions = 3
+ * start_partition = 0
+ * 
+ * output: pair<table, partition_offsets>
+ * table => col 1 {0,3,6,9,12,1,4,7,10,2,5,8,11}
```

Comment:
The word "partition" made me think this function would return multiple tables. The description cleared that up nicely.